.github/workflows/zap_templates.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: ZAP
+
+on:
+    push:
+    pull_request:
+
+jobs:
+    templates:
+        name: ZAP templates generation
+
+        runs-on: ubuntu-latest
+
+        strategy:
+           matrix:
+            node-version: [12.x]
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v1
+              with:
+                node-version: ${{ matrix.node-version }}
+
+            - run: sudo apt-get update
+            - run: sudo apt-get install --fix-missing libpixman-1-dev libcairo-dev libsdl-pango-dev libjpeg-dev libgif-dev
+            - name: Setup ZAP
+              run: |
+                cd third_party/zap/repo/
+                git checkout master
+                git pull
+                npm ci
+                npm run version-stamp
+                npm rebuild canvas --update-binary
+                npm run build-spa
+            - name: Generate
+              run: scripts/tools/zap_generate.sh examples/all-clusters-app/all-clusters-common/all-clusters-app.zap
```

Comment:
how do we verify that the output is correct?

Reply:
The goal here is to test the validity of the templates to prevent people from adding errors inside of them. As for testing the content of the generate files, we need to actually add an extra step to compile those files. This should be done directly inside the workflow for each example once issue #3928 is complete. Once zap generation is a part of the examples CI, then this workflow should be removed since the validity of the templates will be tested elsewhere.

I see this PR as a temporary solution to maintain the integrity of the templates.

Reply:
So where will this place the generated files when it runs?

Reply:
Same path as the *.zap file but under a /gen/ folder. Those file are discarded at the end of the run since the objective of this workflow is simply to test the integrity of the templates and not their output.

Reply:
So they overwrite the existing `gen` files, but then nothing is actually done with that tree after that?

Reply:
exactly